packages/sdk/src/sdk/services/Storage/Storage.ts

```diff
@@ -125,13 +127,24 @@ export class Storage implements StorageService {
       file.name ?? 'blob'
     )
 
+    const signatureData = {
+      timestamp: Date.now()
+    }
+    const sigJson = JSON.stringify(sortObjectKeys(signatureData))
+    const signature = await auth.hashAndSign(sigJson)
+    const signatureEnvelope = {
+      data: sigJson,
+      signature
+    }
+
     // Using axios for now because it supports upload progress,
     // and Node doesn't support XmlHttpRequest
     let response: AxiosResponse<any> | null = null
     const request: AxiosRequestConfig = {
       method: 'post',
       maxContentLength: Infinity,
       data: formData,
+      params: { signature: JSON.stringify(signatureEnvelope) },
```

Comment:
Is query string param an established pattern for this? Why would it not be part of the POST data?

Reply:
CC @rickyrombo @sliptype I think it probably makes more sense to build a pattern into the request layer to sign POST data automatically?

Reply:
Yes I agree, although `Storage` uses axios instead of our custom fetch implementation https://github.com/AudiusProject/audius-protocol/blob/main/packages/sdk/src/sdk/utils/fetch.ts#L3 maybe because all of our other writes are transactions, and there is some axios behavior we want for posts but I'm not sure. Potentially switching Storage to use fetch would be the move here

Reply:
There's probably three places we can put this data:
* query string
* header
* post body

I'm fine supporting any / all of these... but the first two are available for all requests but POST body only for POST.  So if we want signed gets also... I'd advocate for the first two.